### PR TITLE
Mender-client 2.3.1 release docs

### PR DESCRIPTION
**Pull request by oleorhagen**
Cherry pick `mender-client-version` option to `autoversion`, and bump `Mender-client` version to `2.3.1`

### CODE REVIEW

File: 04.Artifacts/10.Yocto-project/01.Building/docs.md [hunk 1]
```diff
@@ -156,11 +156,11 @@ i.e. the top level of the Yocto Project build tree, and run these commands:
 
 <!--AUTOVERSION: "-b % git://github.com/mendersoftware/meta-mender"/meta-mender-->
 ```bash
-git clone -b zeus git://github.com/mendersoftware/meta-mender
+git clone -b dunfell git://github.com/mendersoftware/meta-mender
```
Comment (lluiscampos) [inline 04.Artifacts/10.Yocto-project/01.Building/docs.md:159]: We cannot update the yocto branch.

**Review by lluiscampos — CHANGES_REQUESTED** [review]
Looks good otherwise

**Comment by lluiscampos** [comment]
Nitpick: Could you please rename this PR to something more human-readable? (what does `Mcl` stands for?)

File: 04.Artifacts/10.Yocto-project/01.Building/docs.md (hunk 1 above)
Comment (oleorhagen) [reply 04.Artifacts/10.Yocto-project/01.Building/docs.md:159]: Hmm, why not?

Comment (oleorhagen) [reply 04.Artifacts/10.Yocto-project/01.Building/docs.md:159]: Release ran on dunfell

Comment (lluiscampos) [reply 04.Artifacts/10.Yocto-project/01.Building/docs.md:159]: You are right, we should update it then.

**Review by lluiscampos — APPROVED** [review]

**Review by lluiscampos — APPROVED** [review]

**Pull request merged**